active/Win32/Window_and_Application_API_Spec.md

```diff
@@ -0,0 +1,618 @@
+# Window and Application API Spec
+
+This spec describes updates to the XAML Window and Application APIs to accommodate this move to support Desktop (win32) apps.
+
+## Spec Status
+In review. No scheduled yet for reviewing by the API Board. 
+
+
+## Execution Notes
+
+This API spec contains the the APIs under the following criteria:
+1. Already exists in Window.UI.Xaml.Window. 
+2. Are part of the Mininum Viable Product
+
+There are other proposed APIs proposed that will be part of the future phases. 
+> A Phase doesn't correspond with a product version, it's just a set of Sprints with allocated resources to execute this work.
+
+##   Design Agreement
+- The new Window implementation reuses the current Windows.UI.XAML.Window class and it will be placed under Microsoft.UI.XAML.Window. 
+- Current methods, events, and properties should still work for UWP and add the new Win32 support. 
+- These APIs use different code paths depending on whether WinUI 3 is running on Win32 or UWP. 
+- There are some APIs that don’t work on Win32 or UWP yet in this phase. Those APIs could return **null** or **throw** an exception depending on the case. 
+
+This agreement applies to the Application class too.
+
+## Window Class
+Namespace: Microsoft.UI.Xaml
+
+```CS
+public class Window : IClosable
+```
+
+### **Window management, appearance, and behaviors**
+
+### Window() constructor **[NEW]** 
+The constructor initializes the Window object. Windows are created in the same thread where they are instantiated. Frequently, this thread is the UI Thread.
+
+```CS
+public Window();
+```
+#### Samples
+##### Code Sample
+```CS
+Window window = new Window();
+window.Content = new TextBlock() { Text = "Hello" };
+window.Activate(); 
+```
+##### Markup Sample
+
+```XML
+<Window 
+    x:Class="MainWindow"
+    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+>
+    <Grid><Button Content=”Click” Click=”OnClick”/></Grid>
+</Window>
+```
+```CS
+public partial class MainWindow:Window
+{
+  public MainWindow()
+  {
+	InitializeComponent();
+  }
+}
+```
+##### Multiple Widndows Sample
+Developers can create multi-window experiences with WinUI 3 in the same UI thread. 
+```CS
+Window w = new Window ();
+Button b = new Button() { Content = "Click Me" };
+b.Click += OnClick;
+w.Content = b;
+w.Activate();
+```
+```CS
+private void OnClick(object sender, RoutedEventArgs e)
+{
+   Window w2 = new Window ();
+   w2.Content = new TextBlock(){Text=”Window 2”}
+   w2.Activate();
+}
+```
+##### Create a Window in a separated thread
+To create a Window in a separated thread, you should create the thread first and create a Window in this new Thread. 
+```CS
+private void Button_Click(object sender, RoutedEventArgs e)
+{
+    Thread thread = new Thread(() =>
```

Comment:
>Thread [](start = 4, length = 6)

Does it make sense to have a Xaml worker function / class that makes this more maintainable, abstracts Win32 and UWP, and enables us to service? May we follow-up?

Reply:
This is a good point, but out of scope for 2020. Adding a note to capture this in the backlog.